src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.ParseMapModules.cs

```diff
@@ -0,0 +1,145 @@
+                    long.TryParse(s.AsSpan(pos + 1, end - (pos + 1)), NumberStyles.HexNumber, CultureInfo.InvariantCulture, out long endingAddress))
+                {
+                    return (startingAddress, (int)(endingAddress - startingAddress));
+                }
+            }
+
+            return default;
+        }
+
+        private static bool HasReadAndExecFlags(string s, ref int start, ref int end)
+        {
+            bool sawRead = false, sawExec = false;
+            for (int i = start; i < end; i++)
+            {
+                if (s[i] == 'r')
+                    sawRead = true;
+                else if (s[i] == 'x')
+                    sawExec = true;
+            }
+
+            return sawRead & sawExec;
+        }
+
+        internal static ProcessModuleCollection? ParseMapsModules(int pid)
+        {
+            try
+            {
+                return ParseMapsModulesCore(File.ReadLines(GetMapsFilePathForProcess(pid)));
+            }
+            catch (IOException) { }
+            catch (UnauthorizedAccessException) { }
+
+            return null;
+        }
+
+        private static ProcessModuleCollection ParseMapsModulesCore(IEnumerable<string> lines)
+        {
+            Debug.Assert(lines != null);
+
+            ProcessModule? module = null;
+            ProcessModuleCollection modules = new(capacity: 0);
+            bool moduleHasReadAndExecFlags = false;
+
+            foreach (string line in lines)
+            {
+                // Use a StringParser to avoid string.Split costs
+                var parser = new StringParser(line, separator: ' ', skipEmpty: true);
+
+                // Parse the address start and size
+                (long Start, int Size) addressRange = parser.ParseRaw(TryParseAddressRange);
+
+                if (addressRange == default)
+                {
+                    Commit();
+                    continue;
+                }
+
+                // Parse the permissions
+                bool lineHasReadAndExecFlags = parser.ParseRaw(HasReadAndExecFlags);
+
+                // Skip past the offset, dev, and inode fields
+                parser.MoveNext();
+                parser.MoveNext();
+                parser.MoveNext();
+
+                // Parse the pathname
+                if (!parser.MoveNext())
+                {
+                    Commit();
+                    continue;
+                }
+
+                string pathname = parser.ExtractCurrentToEnd();
+                bool isContinuation = module?.FileName == pathname && (long)module.BaseAddress + module.ModuleMemorySize == addressRange.Start;
+
+                if (isContinuation)
+                {
+                    module!.ModuleMemorySize += addressRange.Size;
+                    continue;
+                }
+
+                Commit();
+                moduleHasReadAndExecFlags |= lineHasReadAndExecFlags;
```

Comment:
`moduleHasReadAndExecFlags` may be more clear if you assign it here:
```
moduleHasReadAndExecFlags = lineHasReadAndExecFlags;
```
Update it under `(isContinuation)`:
```
moduleHasReadAndExecFlags |= lineHasReadAndExecFlags;
```
And remove it `moduleHasReadAndExecFlags = false;` in `Commit`.

Reply:
Simplified as suggested. Thanks! :)